datadog_checks_dev/tests/test_docker.py

```diff
@@ -40,3 +44,60 @@ def test_compose_file(self):
             assert compose_file_active(compose_file) is False
         finally:
             run_command(['docker', 'compose', '-f', compose_file, 'down'], capture=True)
+
+    @pytest.mark.parametrize(
+        "attempts,expected_call_count",
+        [
+            [
+                None,
+                1,
+            ],
+            [
+                0,
+                1,
+            ],
+            [
+                1,
+                1,
+            ],
+            [
+                3,
+                3,
+            ],
+        ],
+    )
+    def test_retry_on_failed_conditions(self, attempts, expected_call_count):
+        compose_file = os.path.join(DOCKER_DIR, "test_default.yaml")
+
+        def condition():
+            condition.call_count += 1
+            raise RetryError("boom!")
+
+        condition.call_count = 0
+
+        try:
+            with pytest.raises(RetryError if attempts is None else tenacity.RetryError):
+                with docker_run(compose_file, attempts=attempts, conditions=[condition]):
+                    pass
+
+            assert condition.call_count == expected_call_count
+        finally:
+            run_command(["docker", "compose", "-f", compose_file, "down"], capture=True)
+
+    def test_retry_condition_failed_only_on_first_run(self):
+        compose_file = os.path.join(DOCKER_DIR, "test_default.yaml")
+
+        def condition():
+            condition.call_count += 1
+
+            if condition.call_count == 1:
+                raise RetryError("boom!")
+
+        condition.call_count = 0
```

Comment:
Consider using a [`mock.Mock`](https://docs.python.org/3.8/library/unittest.mock.html#unittest.mock.Mock) instead (in both tests).

E.g. something roughly like:
```suggestion
        condition = mock.Mock(side_effect=[None, RetryError("boom!")])
```

And then asserting on `condition.call_count` as that's [provided by the mock object](https://docs.python.org/3.8/library/unittest.mock.html#unittest.mock.Mock.call_count) already.

Reply:
Good idea. Updated